File: legacy_backup/api/feishu/records/[record_id].py
from http.server import BaseHTTPRequestHandler
import json
import os
import requests

class handler(BaseHTTPRequestHandler):
    def do_PUT(self):
        content_length = int(self.headers['Content-Length'])
        post_data = self.rfile.read(content_length)
        data = json.loads(post_data)
        
        record_id = self.path.split('/')[-1]
        
        app_id = data.get('app_id', os.environ.get('FEISHU_APP_ID', ''))
        app_secret = data.get('app_secret', os.environ.get('FEISHU_APP_SECRET', ''))
        bitable_app_token = data.get('bitable_app_token')
        bitable_table_id = data.get('bitable_table_id')
        fields = data.get('fields', {})
        
        if not bitable_app_token or not bitable_table_id:
            self.send_response(400)
            self.send_header('Content-type', 'application/json')
            self.end_headers()
            self.wfile.write(json.dumps({'error': '缺少必要参数'}).encode())
            return
        
        if not app_id:
            self.send_response(400)
            self.send_header('Content-type', 'application/json')
            self.end_headers()
            self.wfile.write(json.dumps({'error': '缺少 App ID'}).encode())
            return
        
        try:
            auth_response = requests.post(
                'https://open.feishu.cn/open-apis/auth/v3/tenant_access_token/internal',
                json={
                    'app_id': app_id,
                    'app_secret': app_secret
                }
            )
            
            if auth_response.status_code != 200:
                raise Exception('获取访问令牌失败')
            
            auth_data = auth_response.json()
            access_token = auth_data.get('tenant_access_token')
            
            if not access_token:
                raise Exception('无效的访问令牌')
            
            record_response = requests.put(
                f'https://open.feishu.cn/open-apis/bitable/v1/apps/{bitable_app_token}/tables/{bitable_table_id}/records/{record_id}',
                headers={
                    'Authorization': f'Bearer {access_token}',
                    'Content-Type': 'application/json'
                },
                json={'fields': fields}
            )
            
            if record_response.status_code != 200:
                error_data = record_response.json()
                raise Exception(error_data.get('msg', '更新记录失败'))
            
            self.send_response(200)
            self.send_header('Content-type', 'application/json')
            self.end_headers()
            self.wfile.write(record_response.content)
            
        except Exception as e:
            self.send_response(500)
            self.send_header('Content-type', 'application/json')
            self.end_headers()
            self.wfile.write(json.dumps({'error': str(e)}).encode())

File: api/feishu/records/[record_id].js
export default async function handler(request, response) {
  if (request.method !== 'PUT') {
    return response.status(405).json({ error: 'Method not allowed' });
  }

  try {
    const { record_id } = request.query;
    const body = request.body ? JSON.parse(request.body) : {};
    const app_id = body.app_id || process.env.FEISHU_APP_ID || '';
    const app_secret = body.app_secret || process.env.FEISHU_APP_SECRET || '';
    const bitable_app_token = body.bitable_app_token;
    const bitable_table_id = body.bitable_table_id;
    const fields = body.fields || {};

    if (!bitable_app_token || !bitable_table_id || !record_id) {
      return response.status(400).json({ error: '缺少必要参数' });
    }

    if (!app_id) {
      return response.status(400).json({ error: '缺少 App ID' });
    }

    const auth_response = await fetch(
      'https://open.feishu.cn/open-apis/auth/v3/tenant_access_token/internal',
      {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({
          app_id: app_id,
          app_secret: app_secret
        })
      }
    );

    if (auth_response.status !== 200) {
      throw new Error('获取访问令牌失败');
    }

    const auth_data = await auth_response.json();
    const access_token = auth_data.tenant_access_token;

    if (!access_token) {
      throw new Error('无效的访问令牌');
    }

    const record_response = await fetch(
      `https://open.feishu.cn/open-apis/bitable/v1/apps/${bitable_app_token}/tables/${bitable_table_id}/records/${record_id}`,
      {
        method: 'PUT',
        headers: {
          'Authorization': `Bearer ${access_token}`,
          'Content-Type': 'application/json'
        },
        body: JSON.stringify({ fields: fields })
      }
    );

    if (record_response.status !== 200) {
      const error_data = await record_response.json();
      throw new Error(error_data.msg || '更新记录失败');
    }

    const result = await record_response.json();
    response.status(200).json(result);

  } catch (error) {
    response.status(500).json({ error: error.message });
  }
}
